### PR TITLE
Add hypertrace for recv and send methods in Channel

### PR DESCRIPTION
**Pull request by HDegroote**
Useful for instrumenting hypercore-related apps, to know how often each message type gets sent and received

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -194,16 +197,30 @@ class Channel {
...
         return session._track(m.onmessage(encoding.decode(state), session))
       },
       send (m, session = s) {
         if (session.closed === true) return false
 
+        self.tracer.trace('send', {
+          id: self.id,
+          protocol: self.protocol,
+          type
+        })
```
Comment (freeall) [inline index.js]: On line 197, `s = this` is already defined.

```suggestion
        s.tracer.trace('send', {
          id: s.id,
          protocol: s.protocol,
          type
        })
```

Comment (HDegroote) [reply index.js]: Oops, fixed. Just pushed a commit instead of accepting the suggestion since I also had to update `recv`

File: index.js [hunk 2]
```diff
@@ -199,11 +202,23 @@ class Channel {
...
+
         return session._track(m.onmessage(encoding.decode(state), session))
       },
       send (m, session = s) {
         if (session.closed === true) return false
 
+        s.tracer.trace('send', {
+          id: s.id,
+          protocol: s.protocol,
```
Comment (mafintosh) [inline index.js:217]: super nit but the order should be protocol, id, type for both send and recv

Comment (HDegroote) [reply index.js:217]: Makes sense, fixed

**Pull request merged**